### ISSUE TITLE
Include more schedules in csv file

### ISSUE THREAD

**Issue by joseph-robertson** [NatLabRockies/resstock#735]
Examples:
- heating/cooling seasons (more detailed than the current simple ones)
- bath/kitchen fans
- heating/cooling setpoints
- water heating (e.g., Jeff had included something like this for WH-GEB)

For vacancy, are we zeroing out everything we want to (i.e., all schedules that are currently in the schedules csv file)?

For power outage, would this work?

We *could* move away from rulesets altogether, but there'd be a runtime penalty associated with that.

**Comment by joseph-robertson** [comment]
On setpoints:

Option A
- [ ] Have `BuildResidentialHPXML` (or `ResStockArguments`?) create the setpoint schedule that we want
- [ ] Update `BuildResidentialScheduleFile` a lot like WH-GEB, and pass in the created setpoint schedule

Option B
- [ ] Incorporate setpoint arguments into `BuildResidentialScheduleFile` (they may be simple enough)

**Comment by joseph-robertson** [comment]
@shorowit Seems like we could pursue Option A above if we separated out pieces of the stochastic generator into their own methods. ResStockArguments could call these method(s) and gather relevant information for creating the setpoint schedule csv. The rest of the workflow downstream (in terms of measure calls), therefore, would be untouched. What do you think?

**Comment by shorowit** [comment]
Yes, that is a definite possibility. Though option B may be easier to implement and has some appeal if indeed there are some simple, high-level inputs to construct the 8760 setpoint schedules. @ejhw can you list the inputs that are being proposed?

**Comment by ejhw** [comment]
@shorowit 

- Cooling Setpoint Has Offset Nighttime
- Cooling Setpoint Has Offset Daytime Unoccupied
- Cooling Setpoint Offset Magnitude (e.g., 0, 3, 6, 12 ºF)

- Heating Setpoint Has Offset Nighttime
- Heating Setpoint Has Offset Daytime Unoccupied
- Heating Setpoint Offset Magnitude (e.g., 0, 3, 6, 12 ºF)

It is possible that we could refactor to make it simpler by using 0ºF offset to represent Has Offset = False:
- Cooling Setpoint Offset Nighttime (e.g., 0, 3, 6, 12 ºF)
- Cooling Setpoint Offset Daytime Unoccupied (e.g., 0, 3, 6, 12 ºF)

- Heating Setpoint Offset Nighttime (e.g., 0, 3, 6, 12 ºF)
- Heating Setpoint Offset Daytime Unoccupied (e.g., 0, 3, 6, 12 ºF)

cc @ekpresent

**Comment by joseph-robertson** [comment]
I'm trying to minimally prove this approach out here: https://github.com/NREL/OpenStudio-HPXML/pull/994.

@shorowit @ejhw @ekpresent

**Comment by afontani** [comment]
- heating/cooling seasons: Being built out via the Power Outage PR
- bath/kitchen fans: Being built out via the Power Outage PR
- heating/cooling setpoints: Have open PRs to create the 8760s but hasn't been merged.
- water heating (e.g., Jeff had included something like this for WH-GEB): Has not begun.

**Comment by joseph-robertson** [comment]
@ejhw What was the original motivation for wanting to use ResStock to generate detailed HVAC setpoints? I'm trying to get some traction on #931, which uses (1) occupants/sleeping schedules and (2) new daytime unoccupied / nighttime offset arguments to create the detailed HVAC setpoints.

@afontani @shorowit

**Comment by ejhw** [comment]
@joseph-robertson I think it was 

1) to make HVAC setpoints consistent with occupancy schedules, and 
2) to facilitate GEB setpoint schedule measures that work with the new setpoint diversity added via EULP. 

I think the motivation was for one of @ekpresent 's projects.

**Comment by afontani** [comment]
Related to #931
